repository-ydb-v2/src/main/java/tech/ydb/yoj/repository/ydb/yql/YqlPrimitiveType.java

```diff
@@ -321,17 +331,79 @@ private static void registerYqlType(
+        useRecommendedMappingFor(FieldValueType.STRING, FieldValueType.ENUM);
     }
 
+    /**
+     * @deprecated This method has a misleading name. Call {@link #useLegacyMappingFor(FieldValueType[]) useLegacyMappingFor(STRING, ENUM)} instead.
+     */
+    @Deprecated(forRemoval = true)
     public static void resetStringDefaultTypeToDefaults() {
-        VALUE_DEFAULT_YQL_TYPES.put(FieldValueType.STRING, new ValueYqlTypeSelector(FieldValueType.STRING, PrimitiveTypeId.STRING, null));
-        VALUE_DEFAULT_YQL_TYPES.put(FieldValueType.ENUM, new ValueYqlTypeSelector(FieldValueType.ENUM, PrimitiveTypeId.STRING, null));
+        useLegacyMappingFor(FieldValueType.STRING, FieldValueType.ENUM);
+    }
+
+    /**
+     * Uses the legacy (YOJ 1.0.x) field value type &harr; YDB column type mapping for the specified field value type(s).
+     * If you need to support legacy applications, call {@code useLegacyMappingFor(STRING, ENUM, TIMESTAMP)} before using
+     * any YOJ features.
+     *
+     * @deprecated We <strong>STRONGLY</strong> advise against using the legacy mapping in new projects.
+     * Please call {@link #useRecommendedMappingFor(FieldValueType...) useNewMappingFor(STRING, ENUM, TIMESTAMP)} instead,
+     * and annotate custom-mapped columns with {@link Column &#64;Column} where a different mapping is desired.
+     *
+     * @param fieldValueTypes field value types to use legacy mapping for
+     */
+    @Deprecated
+    public static void useLegacyMappingFor(FieldValueType... fieldValueTypes) {
+        for (var fvt : fieldValueTypes) {
+            switch (fvt) {
+                case STRING, ENUM -> VALUE_DEFAULT_YQL_TYPES.put(fvt, new ValueYqlTypeSelector(fvt, PrimitiveTypeId.STRING, null));
+                case TIMESTAMP -> {
+                    var selector = new YqlTypeSelector(Instant.class, PrimitiveTypeId.INT64, null);
+                    JAVA_DEFAULT_YQL_TYPES.put(Instant.class, selector);
+                    YQL_TYPES.put(selector, new YqlPrimitiveType(Instant.class, PrimitiveTypeId.INT64, INSTANT_SETTER, INSTANT_GETTER));
+                }
+                default -> throw new IllegalArgumentException("There is no legacy mapping for field value type: " + fvt);
+            }
+        }
+    }
+
+    /**
+     * Uses the recommended field value type &harr; YDB column type mapping for the specified field value type(s).
+     * <p>
+     * In new projects, we <strong>STRONGLY</strong> advise that you call {@code useNewMappingFor(STRING, ENUM, TIMESTAMP)}
+     * before using any YOJ features. This will eventually become the default mapping, and the call will become a no-op and
+     * mighe even be removed.
+     *
+     * @param fieldValueTypes field value types to use the new mapping for
+     */
+    @ExperimentalApi(issue = "https://github.com/ydb-platform/yoj-project/issues/20")
+    public static void useRecommendedMappingFor(FieldValueType... fieldValueTypes) {
```

Comment:
nitpick: looks little bit strange: useLegacyMappingFor is deprecated, but useRecommendedMappingFor is experimental. Where is a correct way to set mappings?